app/src/main/java/com/concordium/wallet/data/util/CurrencyUtil.kt

```diff
@@ -27,8 +28,14 @@ object CurrencyUtil {
         return formatGTU(value, withGStroke, decimals)
     }
 
+    fun setGstroke(gStroke: String){
+        this.gStroke = gStroke
+    }
+
     fun formatGTU(value: Long, withGStroke: Boolean = false, decimals: Int = 6): String {
-        val gStroke = App.appContext.getString(R.string.app_gstroke)
+        if(withGStroke && gStroke == null){
+            gStroke = App.appContext.getString(R.string.app_gstroke)
+        }
```

Comment:
I really don't like this lazy initialization logic. You shouldn't have to molest your code to make it testable. There must be a better way, but I'm not sure what it is. Maybe the context should have been a singleton that doesn't require the full app to be initialized to access it. Then the roboelectric runner would probably work (I tried that but it failed with `AndroidKeyStore not found` because it tried to init the entire app which apparently it can't).

Let's do this for now and rework it once we find figure out what the proper solution is.

Reply:
Ok